src/mpid/ch4/netmod/ofi/ofi_rma.h

```diff
@@ -330,10 +330,43 @@ static inline int MPIDI_OFI_do_put(const void *origin_addr,
     struct iovec *originv;
     struct fi_rma_iov *targetv;
     MPIDI_OFI_seg_state_t p;
+    struct iovec iov;
+    struct fi_rma_iov riov;
+    int origin_contig, target_contig;
+    size_t origin_bytes, target_bytes;
+    MPI_Aint origin_true_lb, target_true_lb;
 
     MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDI_OFI_DO_PUT);
     MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDI_OFI_DO_PUT);
 
+    offset = target_disp * MPIDI_OFI_winfo_disp_unit(win, target_rank);
+    msg.desc = NULL;
+    msg.addr = MPIDI_OFI_comm_to_phys(win->comm_ptr, target_rank);
+    msg.context = NULL;
+    msg.data = 0;
+
+    MPIDI_Datatype_check_contig_size_lb(target_datatype, target_count, target_contig, target_bytes,
+                                        target_true_lb);
+    MPIDI_Datatype_check_contig_size_lb(origin_datatype, origin_count, origin_contig, origin_bytes,
+                                        origin_true_lb);
```

Comment:
In case of `MPI_Put`, we see calls to `MPIDI_Datatype_check_contig_size_lb` once in `NM_mpi_put` and another time here. A clever compiler might be able to wipe out the redundant code though.

One thing I could think is to add `origin_contig`, `target_contig`, etc. for `do_put`'s parameter list.

Reply:
Yes we need `origin_contig` and `target_contig` for `do_put` so adding them for `do_put`'s parameter list seems to be the only way. By that, we would need to call `MPIDI_Datatype_check_contig_size_lb` in `MPIDI_NM_mpi_rput` too.

Reply:
I think it is harmless to add two parameters `origin_contig` and `target_contig` for `ofi_do_put`, same for `do_get`.